maintenance/wikia/AddDiscussionsToNavigation.php

```diff
@@ -0,0 +1,152 @@
+<?php
+
+/**
+ * Maintenance script to update navigation with Discussion links.
+ * @author Michal 'slayful' Turek
+ */
+
+ini_set('display_errors', 'stderr');
+ini_set('error_reporting', E_NOTICE);
+
+require_once(dirname(__FILE__) . '/../Maintenance.php');
+
+/**
+ * Class AddDiscussionsToNavigation
+ */
+class AddDiscussionsToNavigation extends Maintenance
+{
+
+	const VAR_NAME = 'wgOasisGlobalNavigation';
+	const NAVIGATION_ELEMENT = '**Special:DiscussionsNavigation|Discussions';
+	const FORUM_NAVIGATION_ELEMENT = '**Special:Forum|Forum';
+	const REASON = 'SOC-2816 add Discussion navigation to the navigation';
+	const SPLIT = "\r\n";
+
+	private $dryRun = false;
+	private $verbose = false;
+	private $rollback = false;
+
+	public function __construct() {
+		parent::__construct();
+		$this->mDescription = "Update " . self::VAR_NAME . " with Discussions navigation "
+			. self::NAVIGATION_ELEMENT;
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Use "static" keyword instead of "self". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS2037)